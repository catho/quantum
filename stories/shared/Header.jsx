import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Unstyled } from '@storybook/addon-docs';
import { theme } from '../../components/shared';

import ArrowLeft from '../../.storybook/static/arrow-left.png';
import ArrowRight from '../../.storybook/static/arrow-right.png';

const {
  spacing: { large },
  colors: { secondary, neutral },
} = theme;

const HeaderWrapper = styled.header`
  position: relative;
  display: flex;
  flex-direction: column;
  align-items: flex-start;

  margin-bottom: ${large}px;

  background: linear-gradient(
    101.73deg,
    #0023a0 25.56%,
    ${secondary[700]} 99.08%
  );
  z-index: 0;

  &::before,
  &::after {
    content: '';
    position: absolute;
    width: 100%;
    height: 100%;
  }

  &::before {
    background: url(${ArrowLeft}) no-repeat left top;
  }

  &::after {
    background: url(${ArrowRight}) no-repeat right bottom;
  }

  & > div {
    width: 100%;
    z-index: 1;
  }

  h1,
  p {
    color: ${neutral[0]};
  }
`;

const Container = styled.div`
  margin: 0 auto;
  max-width: 87%;
  padding-bottom: 90px;
`;

const Title = styled.h1`
  align-items: center;
  color: #294661;
  display: flex;
  flex-direction: row;
  font-size: 40px;
  font-weight: 700;
  letter-spacing: -1px;
  line-height: 60px;
  margin: 0 0 30px 0;
  padding: 90px 0 0 0;
  text-align: left;

  ${({ as }) => as === 'h2' && `font-size: 30px;`}
  ${({ as }) => as === 'h3' && `font-size: 24px;`}
`;

const Header = ({ title, children }) => (
  <Unstyled>
    <HeaderWrapper>
      <Container>
        <Title>{title}</Title>
        <p>{children}</p>
      </Container>
    </HeaderWrapper>
  </Unstyled>
);

Header.defaultProps = {
  children: undefined,
};

Header.propTypes = {
  title: PropTypes.string.isRequired,
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]),
};

export default Header;
