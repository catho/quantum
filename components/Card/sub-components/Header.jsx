import styled from 'styled-components';
import PropTypes from 'prop-types';

import { spacing } from '../../shared/theme';

const Header = styled.header`
  display: flex;

  ${({
    theme: {
      spacing: { medium },
    },
  }) => `
    padding: ${medium}px ${medium}px 0;

    > * {
      margin-right: ${medium}px;
    }

    > *:last-child {
      margin-right: 0;
    }
  `}
`;

Header.propTypes = {
  theme: PropTypes.shape({
    spacing: PropTypes.object,
  }),
};

Header.defaultProps = {
  theme: {
    spacing,
  },
};

Header.displayName = 'Card.Header';

export default Header;
