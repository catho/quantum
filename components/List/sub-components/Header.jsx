import PropTypes from 'prop-types';
import styled from 'styled-components';
import { spacing } from '../../shared/theme';

const Title = styled.span`
  ${({
    theme: {
      spacing: { xxsmall },
    },
  }) => `
  margin: ${xxsmall}px 0;
`}
`;

Title.displayName = 'Header';

const Header = ({
  children,
  theme = {
    spacing,
  },
}) => <Title theme={theme}>{children}</Title>;

Header.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node,
  ]).isRequired,
  theme: PropTypes.shape({
    spacing: PropTypes.object,
  }),
};

Header.displayName = 'List.Header';

export default Header;
