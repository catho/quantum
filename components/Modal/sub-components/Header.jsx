import styled from 'styled-components';
import PropTypes from 'prop-types';
import Card from '../../Card';
import { spacing } from '../../shared/theme';

const Header = styled(Card.Header)`
  padding: ${({
    theme: {
      spacing: { medium },
    },
  }) => `${medium}px ${medium}px 0`};
`;

Header.propTypes = {
  theme: PropTypes.shape({
    spacing: PropTypes.object,
  }),
};

Header.defaultProps = {
  theme: { spacing },
};

export default Header;
