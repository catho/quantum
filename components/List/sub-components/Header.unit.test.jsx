import React from 'react';
import { mount } from 'enzyme';
import toJson from 'enzyme-to-json';
import Header from './Header';

describe('<Header />', () => {
  it('Should match the snapshot', () => {
    const wrapper = mount(<Header>Header</Header>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
