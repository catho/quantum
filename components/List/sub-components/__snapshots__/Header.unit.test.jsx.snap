// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> Should match the snapshot 1`] = `
.c0 {
  margin-top: 4px;
  margin-bottom: 4px;
}

<List.Header>
  <Header>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "Header__Title-e8ruju-0",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "margin-top:4px;margin-bottom:4px;",
            ],
          },
          "displayName": "Header",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "Header__Title-e8ruju-0",
          "target": "span",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <span
        className="c0"
      >
        Header
      </span>
    </StyledComponent>
  </Header>
</List.Header>
`;
